core/miner/miner.hpp

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CPP_FILECOIN_MINER_MINER_HPP
+#define CPP_FILECOIN_MINER_MINER_HPP
+
+#include <boost/asio.hpp>
+
+#include "api/api.hpp"
+#include "common/outcome.hpp"
+#include "miner/storage_fsm/sealing.hpp"
+#include "sector_storage/manager.hpp"
+#include "primitives/address/address.hpp"
+#include "primitives/stored_counter/stored_counter.hpp"
+#include "storage/chain/chain_store.hpp"
+
+namespace fc::miner {
+  using api::Api;
+  using mining::Sealing;
+  using primitives::Counter;
+  using primitives::address::Address;
+  using storage::blockchain::ChainStore;
+  using sector_storage::Manager;
+
+  class Miner {
+   public:
+    Miner(std::shared_ptr<Api> api,
+          Address miner_address,
+          Address worker_address,
+          std::shared_ptr<ChainStore> chain_store,
+          std::shared_ptr<Counter> counter,
+          std::shared_ptr<Manager> sector_manager,
+          std::shared_ptr<boost::asio::io_context> context);
+
+    outcome::result<void> run();
+
+    void stop();
+
+    /**
+     * Checks miner worker address
+     * @return error if worker address is incorrect
+     */
+    outcome::result<void> runPreflightChecks();
+
+   private:
+    std::shared_ptr<Api> api_;
+    Address miner_address_;
+    Address worker_address_;
+    std::shared_ptr<Sealing> sealing_;
+    std::shared_ptr<ChainStore> chain_store_;
```

Comment:
Separate "miner" process can't access `ChainStore`, `EventsImpl` should use `Api.ChainNotify`

Reply:
It's related to subscribe to ChainChanges. We have to decide about how to implement it.

Reply:
done